src/Cli/dotnet/Commands/Test/TestingPlatformCommand.cs

```diff
@@ -29,50 +29,63 @@ public int Run(ParseResult parseResult)
         int exitCode = ExitCode.Success;
         try
         {
-            ValidationUtility.ValidateMutuallyExclusiveOptions(parseResult);
+            exitCode = RunInternal(parseResult);
+        }
+        catch
+        {
+            // Make sure we don't return success when we caught error, but also don't overwrite existing error state.
+            if (exitCode == ExitCode.Success)
+            {
+                exitCode = ExitCode.GenericFailure;
+            }
+        }
```

Comment:
This is very concerning. If we didn't catch here, does the exception propagate up to `Main` and crashes the dotnet process?

If it already propagates up to Main correctly, then I think we shouldn't catch here and let the runtime do its job of printing exception details etc.

Reply:
done, removed the catch. We write the summary exit code message only when the test run was able to complete, which should give us one of the know testing platform exit codes, and not when an exception was thrown